### PR TITLE
Fix all clippy lints

### PR DESCRIPTION
**Pull request by e00E**

### CODE REVIEW

File: src/unstructured.rs [hunk 1]
```diff
@@ -405,11 +405,9 @@ impl<'a> Unstructured<'a> {
...
     pub fn fill_buffer(&mut self, buffer: &mut [u8]) -> Result<()> {
         let n = std::cmp::min(buffer.len(), self.data.len());
-        for i in 0..n {
-            buffer[i] = self.data[i];
-        }
-        for i in self.data.len()..buffer.len() {
-            buffer[i] = 0;
+        buffer[..n].copy_from_slice(&self.data[..n]);
+        for byte in buffer[n..].iter_mut() {
+            *byte = 0;
```
Comment (e00E) [inline src/unstructured.rs:410]: This is the only non obvious change. Clippy didn't suggest exactly this solution but it complained about the previous loop so I changed it like this. Imo this makes it more clear that the rest of the buffer is set to 0.

File: src/lib.rs [hunk 2]
```diff
@@ -347,12 +347,12 @@ impl_arbitrary_for_floats! {
 impl<'a> Arbitrary<'a> for char {
     fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
         use std::char;
-        const CHAR_END: u32 = 0x0011_000;
+        const CHAR_END: u32 = 0x0001_1000;
```
Comment (Manishearth) [inline src/lib.rs:350]: Isn't the correct number here `0x11_0000`? cc @fitzgen

The highest Unicode code point is `0x10_FFFF`

Looks like clippy found a bug?

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
Merging this but I'll file a followup for the char thing

File: src/lib.rs (hunk 2 above)
Comment (fitzgen) [reply src/lib.rs:350]: I was definitely cribbing from some other code and I guess I mistranslated it, whoops! :grimacing:

**Pull request merged**